### PR TITLE
Add I2C error verifications for lsm303

### PR DESCRIPTION
**Pull request by alexandruradovici**
### Pull Request Overview

This pull request adds i2c error verification for the lsm303 drivers.

### Testing Strategy

This pull request was tested using a microbit v2.

### TODO or Help Wanted

This pull request requires #2538 

### Documentation Updated

- [x] No updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: capsules/src/lsm303agr.rs [hunk 1]
```diff
@@ -216,8 +219,9 @@ impl<'a> Lsm303agrI2C<'a> {
             // turn on i2c to send commands
             buf[0] = 0x0F;
             self.i2c_magnetometer.enable();
-            // TODO verify errors
-            let _ = self.i2c_magnetometer.write_read(buf, 1, 1);
+            if let Err((error, buf)) = self.i2c_magnetometer.write_read(buf, 1, 1) {
+                self.command_complete(buf, Err(error));
```
Comment (hudson-ayers) [inline capsules/src/lsm303agr.rs:223]: Calling a callback in a downcall is a bug in Tock, this error would need to be delivered via a deferred call (see https://github.com/tock/tock/blob/master/doc/reference/trd-hil-design.md#rule-1-dont-make-synchronous-callbacks). Same goes for the other `command_complete()` calls in this file that can happen immediately in response to a downcall

File: capsules/src/lsm303dlhc.rs [hunk 2]
```diff
@@ -221,8 +221,9 @@ impl<'a> Lsm303dlhcI2C<'a> {
             // turn on i2c to send commands
             buf[0] = 0x0F;
             self.i2c_magnetometer.enable();
-            // TODO verify errors
-            let _ = self.i2c_magnetometer.write_read(buf, 1, 1);
+            if let Err((error, buf)) = self.i2c_magnetometer.write_read(buf, 1, 1) {
+                self.command_complete(buf, Err(error));
```
Comment (hudson-ayers) [inline capsules/src/lsm303dlhc.rs:225]: same issue here

**Review by hudson-ayers — CHANGES_REQUESTED** [review]

File: capsules/src/lsm303agr.rs (hunk 1 above)
Comment (alexandruradovici) [reply capsules/src/lsm303agr.rs]: Done

File: capsules/src/lsm303dlhc.rs (hunk 2 above)
Comment (alexandruradovici) [reply capsules/src/lsm303dlhc.rs]: Done

**Review by hudson-ayers — DISMISSED** [review]

File: capsules/src/lsm303agr.rs [hunk 3]
```diff
@@ -374,7 +429,7 @@ impl i2c::I2CClient for Lsm303agrI2C<'_> {
                 self.i2c_accelerometer.disable();
                 self.state.set(State::Idle);
                 if self.config_in_progress.get() {
-                    self.set_magneto_data_rate(self.mag_data_rate.get());
+                    let _ = self.set_magneto_data_rate(self.mag_data_rate.get());
```
Comment (phil-levis) [inline capsules/src/lsm303agr.rs:432]: Could this cause things to hang if the I2C is busy for some reason? I'm a little wary of the potential error being discarded. If you have reason to think that this should not happen, then please add a comment explaining why.

File: capsules/src/lsm303agr.rs [hunk 4]
```diff
@@ -436,7 +491,7 @@ impl i2c::I2CClient for Lsm303agrI2C<'_> {
                 self.i2c_magnetometer.disable();
                 self.state.set(State::Idle);
                 if self.config_in_progress.get() {
-                    self.set_range(self.mag_range.get());
+                    let _ = self.set_range(self.mag_range.get());
```
Comment (phil-levis) [inline capsules/src/lsm303agr.rs:494]: Same as above. Shouldn't an error result in an upcall with an error?

**Review by phil-levis — COMMENTED** [review]
Two error cases seem to not be handled; if they can't occur, it would be nice to have a comment that explains why.

File: capsules/src/lsm303agr.rs (hunk 3 above)
Comment (alexandruradovici) [reply capsules/src/lsm303agr.rs]: Thank you for pointing this out.

File: capsules/src/lsm303agr.rs (hunk 4 above)
Comment (alexandruradovici) [reply capsules/src/lsm303agr.rs]: I updated it.

**Review by bradjc — APPROVED** [review]

**Review by hudson-ayers — APPROVED** [review]

**Review by alistair23 — APPROVED** [review]

**Pull request merged**